src/test/java/core/basesyntax/FruitTransactionServiceTest.java

```diff
@@ -0,0 +1,101 @@
+            new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                    "banana", 350);
+    private static final FruitTransaction BANANA_FRUIT_PURCHASE =
+            new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                    "banana", 100);
+    private static final FruitTransaction BANANA_FRUIT_RETURN =
+            new FruitTransaction(FruitTransaction.Operation.RETURN,
+                    "banana", 50);
+    private static final int BANANA_QUANTITY_AFTER_PROCESSING = 550;
+    private static final int ORANGE_QUANTITY_AFTER_PROCESSING = 190;
+    private static final String BANANA = "banana";
+    private static final String ORANGE = "orange";
+    private static FruitTransactionService fruitTransactionService;
+    private static FruitShopDao fruitShopDao;
+    private static FruitStrategy fruitStrategy;
+    private static Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    @BeforeAll
+    static void initialize_variables() {
+        operationHandlers = new HashMap<>();
+        ListService listService = new ListServiceImpl();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE,
+                new BalanceHandlerImpl(listService));
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseHandlerImpl(listService));
+        operationHandlers.put(FruitTransaction.Operation.RETURN,
+                new ReturnHandlerImpl(listService));
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyHandlerImpl(listService));
+        fruitShopDao = new FruitShopDaoImpl();
+        fruitStrategy = new FruitStrategyImpl(operationHandlers);
+        fruitTransactionService = new FruitTransactionServiceImpl(fruitShopDao, fruitStrategy);
+    }
+
+    @Test
+    void fruitTransactionService_processTransaction_nullNotOk() {
+        Assert.assertThrows(RuntimeException.class,
+                () -> fruitTransactionService.processTransactions(null));
+    }
+
+    @Test
+    void fruitTransactionService_processTransaction_passFruitListOk() {
+        List<FruitTransaction> fruitsToProcess = new ArrayList<>();
+        fruitsToProcess.add(ORANGE_FRUIT_BALANCE);
+        fruitsToProcess.add(ORANGE_FRUIT_SUPPLY);
+        fruitsToProcess.add(ORANGE_FRUIT_PURCHASE);
+        fruitsToProcess.add(ORANGE_FRUIT_RETURN);
+        fruitsToProcess.add(BANANA_FRUIT_BALANCE);
+        fruitsToProcess.add(BANANA_FRUIT_SUPPLY);
+        fruitsToProcess.add(BANANA_FRUIT_PURCHASE);
+        fruitsToProcess.add(BANANA_FRUIT_RETURN);
+        Map<String, Integer> fruitsAfterProcessing =
+                fruitTransactionService.processTransactions(fruitsToProcess);
+        Assertions.assertEquals(BANANA_QUANTITY_AFTER_PROCESSING,
+                fruitsAfterProcessing.get(BANANA));
+        Assertions.assertEquals(ORANGE_QUANTITY_AFTER_PROCESSING,
+                fruitsAfterProcessing.get(ORANGE));
+    }
```

Comment:
You should break it down into smaller tests, and test different combinations of operations.